component/s3/file/s3.go

```diff
@@ -0,0 +1,213 @@
+package s3
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	aws_config "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/grafana/agent/component"
+	"golang.org/x/net/context"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "s3.file",
+		Args:    Arguments{},
+		Exports: Exports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+}
+
+// S3 handles reading content from a file located in S3
+type S3 struct {
+	mut        sync.Mutex
+	opts       component.Options
+	args       Arguments
+	health     *component.Health
+	watcher    *watcher
+	updateChan chan []byte
+	errorChan  chan error
+	content    string
+	cancel     context.CancelFunc
```

Comment:
nit: can you group these with what actually needs to be protected by the mutex and what doesn't need to?